### ISSUE TITLE
reserved identifier violation

### ISSUE THREAD

**Issue by elfring** [searchivarius/nmslib#257]
I would like to point out that identifiers like “[`_CMD_OPTIONS_H_`](https://github.com/searchivarius/nmslib/blob/b0d05cac24b51cdfac498bac6a09460428fe09f3/similarity_search/include/cmd_options.h#L18 "Update candidate")” and “[`__HASH_TABLE_HELPERS_H__`](https://github.com/searchivarius/nmslib/blob/24d7b4c71e269b518b1e1824ab455fe02f637d47/similarity_search/include/falconn/core/hash_table_helpers.h#L1 "Another update candidate")” [do eventually not fit](https://www.securecoding.cert.org/confluence/display/cplusplus/DCL51-CPP.+Do+not+declare+or+define+a+reserved+identifier#DCL51-CPP.Donotdeclareordefineareservedidentifier-NoncompliantCodeExample%28HeaderGuard%29 "Do not declare an identifier which is reserved for the compiler implementation.") to the expected naming convention of the C++ language standard.
Would you like to adjust your selection for unique names?

**Comment by searchivarius** [comment]
@elfring I agree, perhaps, implementing #175 (although not exactly standard) would be a good way to get rid of these non-compliant defines.
